docs/samples/client/csharp/oauth/readme.md

```diff
@@ -0,0 +1,38 @@
+# Authentication using AuthenticationTokenProvider
+
+This readme is intended to demonstrate using the authentication token provider.
+
+## Table of contents
+
+- [Creating a client with OAuth Token](#create-custom-client-constructor-to-support-oauth-token)
+- Example implementation of AuthenticationTokenProvider
+- Example usage of AuthenticationTokenProvider and Custom Client
+
+### Create custom client constructor to support OAuth token
+
+This constructor initializes a new instance of the `UnbrandedTypeSpecClient` class and sets a client pipeline with OAuth2 bearer token authentication using the provided `AuthenticationTokenProvider` and endpoint URI.
+
+```csharp
+using System.ClientModel;
+using System.ClientModel.Primitives;
+
+/// <summary>
+/// Customize client to add custom constructor.
+/// </summary>
+public partial class UnbrandedTypeSpecClient
```

Comment:
Lets use the SampleTypeSpecClient naming in the md file now even if we will rename the cs files later.